datastore.go

```diff
@@ -90,11 +106,31 @@ func NewDatastore(path string, options *Options) (*Datastore, error) {
 		return nil, err
 	}
 
-	return &Datastore{
+	ds := &Datastore{
 		DB:             kv,
 		closing:        make(chan struct{}),
 		gcDiscardRatio: gcDiscardRatio,
-	}, nil
+		maxGcDuration:  maxGcDuration,
+		gcInterval:     gcInterval,
+	}
+
+	// schedule periodic GC till db is closed
+	go ds.periodicGC()
+
+	return ds, nil
+}
+
+func (d *Datastore) periodicGC() {
+	for {
+		select {
+		case <-time.After(d.gcInterval):
```

Comment:
isn't it more natural to have a `Ticker` ? Or a timer that can be stopped when before closing.

(also depends if the interval should be a fixed interval or actually a countdown since the last GC, in which case the user should not expect GCs happening every 15 mins but after 15 mins the previous GC finished (that need to be more clarified in docs)).

Reply:
@hsanjuan  I think waiting for `gcInterval` between two iterations makes more sense for a use case like GC.   I'll add some documentation to that regard.